### PR TITLE
Use formspec version 6 in the main menu

### PR DESCRIPTION
**Pull request by rubenwardy**
This makes all the tabs in the mainmenu use formspec version 6 and real coordinates. To do this, the local tab has been updated to work with real coordinates

This is a prerequisite to replacing the settings tab with a settings button, part of #13476 

image

## To do

This PR is Ready for Review 

- [x] Make fields on the left longer, to take up available space minus a margin

## How to test

* Open main menu 
* Test that the Start Game tab works as expected

### CODE REVIEW

**Review by grorp — CHANGES_REQUESTED** [review]
I tested this on my Android device with default Minetest scaling settings. Here is a screenshot:

screenshot new

The left side of the tab looks very crowded, although there is still free space at the bottom. The checkboxes even overlap each other: If you zoom in, you can see that they are wider than they are tall. I think more vertical spacing is needed.

This was slightly better with the old version of the tab:

screenshot old

In particular, the checkboxes did not overlap, but were nearly exactly aligned.

Looks good to me otherwise, but I can't look at the code right now. (The "Open Settings" button got even more ugly, but that doesn't matter for obvious reasons.)

**Comment by rubenwardy** [comment]
I forgot to increase the spacing between the checkboxes to account for the move to real coords, done now

**Comment by grorp** [comment]
What about this? https://github.com/grorp/minetest/commit/4c6f28a6101b40bc00de2408c398c05eb48d4a7c

screenshot after proposed changes


It makes the left side use the full available vertical space again, i.e. the bottom of the last field is at `7.1 - 0.375`.

**Comment by rubenwardy** [comment]
Looks good, applied that

File: builtin/fstk/buttonbar.lua [hunk 1]
```diff
@@ -22,27 +22,28 @@ local function buttonbar_formspec(self)
...
 		local btn_pos = {}
 
 		if self.orientation == "horizontal" then
 			btn_pos.x = self.pos.x + --base pos
-			(i - self.startbutton) * self.btn_size +       --button offset
+			(i - self.startbutton) * self.btn_size * 1.25 +       --button offset
 			self.btn_initial_offset
 		else
-			btn_pos.x = self.pos.x + (self.btn_size * 0.05)
+			btn_pos.x = self.pos.x + self.size.x / 2 + self.btn_size / 2
```
Comment (grorp) [inline builtin/fstk/buttonbar.lua:38]:
```suggestion
			btn_pos.x = self.pos.x + self.size.x / 2 - self.btn_size / 2
```

**Review by grorp — APPROVED** [review]
LGTM apart from that.

**Pull request merged**